### PR TITLE
🌱 E2E tests CentOS support

### PR DESCRIPTION
**Pull request by wgslr**
**What this PR does / why we need it**:

This PR makes the e2e tests pass on a CentOS host by adding a modified template file.

This PR builds on top of #218, so the diff is currently larger than just the relevant changes.

### CODE REVIEW

**Comment by wgslr** [comment]
Needs to be merged after #218 
/hold

**Review by namnx228 — COMMENTED** [review]
LGTM

**Review by lentzi90 — APPROVED** [review]

File: hack/e2e/environment.sh
Comment (furkatgofurov7) [inline hack/e2e/environment.sh:15]:
```suggestion
        echo "Supported OS distros for the host are: CentOS Stream 8 or RHEL8 or Ubuntu20.04"
```

File: Makefile
Comment (fmuyassarov) [inline Makefile:127]:
```suggestion
      @for image in $(SOME_LIST); do $(CONTAINER_RUNTIME) pull; done
```

Comment (fmuyassarov) [reply Makefile:127]: not necessarily important, so I leave it to you do decide which one to choose :)

File: hack/e2e/environment.sh
Comment (furkatgofurov7) [inline hack/e2e/environment.sh:40]: We can drop these I believe, only `OS` provided should be enough to find out both [CONTAINER_RUNTIME](https://github.com/metal3-io/metal3-dev-env/blob/master/lib/common.sh#L58-L62) and [EPHEMERAL_CLUSTER](https://github.com/metal3-io/metal3-dev-env/blob/master/lib/common.sh#L201-L206) vars

File: hack/e2e/environment.sh
Comment (furkatgofurov7) [inline hack/e2e/environment.sh:44]: Same

File: templates/test/cluster-template-prow-ha.yaml
Comment (furkatgofurov7) [inline templates/test/cluster-template-prow-ha.yaml:207]: why are we reverting it back? if that is the case, you must get rid of the `IMAGE_FORMAT` var from environment.sh

File: templates/test/cluster-template-prow-ha.yaml
Comment (furkatgofurov7) [inline templates/test/cluster-template-prow-ha.yaml:303]: same

File: templates/test/cluster-template-prow-ha-centos.yaml
Comment (furkatgofurov7) [inline templates/test/cluster-template-prow-ha-centos.yaml:244]: why not to parameterize?

File: templates/test/cluster-template-prow-ha-centos.yaml
Comment (furkatgofurov7) [inline templates/test/cluster-template-prow-ha-centos.yaml:340]: same

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Comment by furkatgofurov7** [comment]
/hold

Let's wait for e2e centos job to be in place to test these changes in CI.

File: templates/test/cluster-template-prow-ha.yaml
Comment (wgslr) [reply templates/test/cluster-template-prow-ha.yaml:207]: Yup, I realized that I need to remove IMAGE_FORMAT from environment.sh as well, just didn't push it yet.
As for why: since I use the variables `IMAGE_RAW_URL` and `IMAGE_RAW_CHECKSUM`, the image can be always assumed to use the raw format. Changing this will require modifying the template here anyway, so I think it's better not to give a false sense of the image format being changeable by `environment.sh` alone.

**Comment by wgslr** [comment]
test-v1a5-centos-e2e

**Comment by wgslr** [comment]
Centos CI job is now present
/unhold
/test-v1a5-centos-e2e
/test-v1a5-e2e
/test-integration

**Review by furkatgofurov7 — COMMENTED** [review]
/approve

File: Makefile
Comment (Xenwar) [inline Makefile:128]:
```suggestion
		$(CONTAINER_RUNTIME) pull $image; \
```

Comment (wgslr) [reply Makefile:128]: This is necessary to espace $ in the Makefile

**Pull request merged**
